kubernetes/nodecollection.go

```diff
@@ -281,44 +318,85 @@ func (s sourceName) cadvisorMetrics() string {
 	return fmt.Sprintf("%s-cadvisor_metrics-%s", s.prefix, s.nodeName)
 }
 
-// retrieveNodeData fetches summary and container data from the node
-func retrieveNodeData(nd nodeFetchData, c raw.Client, mask EndpointMask, api nodeAPI) error {
+// retrieveNodeData fetches summary and container data for the node
+func retrieveNodeData(nd nodeFetchData, config KubeAgentConfig, ns NodeSource, n v1.Node) error {
+	connectionMethods := connectionOptions(config, n, nd, ns)
 	source := sourceName{
 		prefix:   nd.prefix,
 		nodeName: nd.nodeName,
 	}
-	var err error
-
-	if mask.Available(NodeStatsSummaryEndpoint) {
-		// fetch stats/summary data
-		log.Debug("Fetching data from /stats/summary endpoint")
-		_, err = c.GetRawEndPoint(http.MethodGet, source.summary(), nd.workDir, api.statsSummary(), nil, true)
+	toFetch := map[Endpoint]bool{
```

Comment:
Does this mean we're going to try and fetch all of the endpoints automatically every time?

Reply:
Yes, this is consistent with the previous behavior as well